dlp/deidentifyWithTimeExtraction.js

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// sample-metadata:
+//  title: De-identify with time extraction
+//  description: De-identify sensitive data in a string by replacing it with a given time config.
+//  usage: node deidentifyWithTimeExtraction.js my-project string
+function main(projectId, string) {
+  // [START dlp_deidentify_time_extract]
+  // Imports the Google Cloud Data Loss Prevention library
+  const DLP = require('@google-cloud/dlp');
+
+  // Instantiates a client
+  const dlp = new DLP.DlpServiceClient();
+
+  // The project ID to run the API call under
+  // const projectId = 'my-project';
+
+  // The string to de-identify
+  // const string = 'My BirthDay is on 9/21/1976';
+
+  async function deidentifyWithTimeExtraction() {
```

Comment:
Please align the example with https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/dlp/snippets/src/main/java/dlp/snippets/DeIdentifyWithTimeExtraction.java
It uses a table.